source/dagger/gameplay/ping_pong/pingpong_ball.cpp

```diff
@@ -60,7 +60,15 @@ void PingPongBallSystem::Run()
                 if (std::abs(collisionSides.y) > 0)
                 {
                     ball.speed.y *= -1;
-                }
+                }  
+            }
+
+            if (Engine::Registry().has<ControllerMapping>(col.colidedWith) && ball.isMalicious == true) {
```

Comment:
it's better to check light things first:
if (ball.isMalicious == true && Engine::Registry().has<ControllerMapping>(col.colidedWith)) {

if ball.isMalicious isn't true then you wouldn't need to  check further things, and c++ is doing this for you.

Reply:
Oh , yeah... You're right. I rushed it a bit and didn't notice . Will fix that too

Reply:
Swapped the conditions as you sugested